lib/collection.js

```diff
@@ -2253,6 +2253,11 @@ var findAndModify = function(self, query, sort, doc, options, callback) {
   // Execute the command
   self.s.db.command(queryObject, finalOptions, function(err, result) {
     if (err) return handleCallback(callback, err, null);
+
+    if (result && result.value) {
```

Comment:
Like above, we can condense the `if` statement and function call to:

```js
if (result && result.value && typeof self.s.options.deserialize === 'function) {
  result.value = self.s.options.deserialize(result.value);
}
```

And we can eliminate the use of `deserializeDocument`